Implement loyalty api

GabijaG opened the pull request.

PauliusLu commented on VisualRiders.PointOfSale.Project/Dto/CreateUpdateReservationDto.cs:
```diff
@@ -0,0 +1,23 @@
+
+public class CreateUpdateReservationDto
+{
+    public Guid Id { get; set; }
```
As far I could see, none of our CreateUpdate Dtos had Id. Are you sure you need id here? Is it not auto-generated on creation and provided from route on update?

gabkeib replied:
Good point. Id is now removed

PauliusLu reviewed (commented):
Seems ok, 1 comment

Commit:
Implement Loyalties api

A Note: in the database model discountId type is long. Everywhere else for id - int type is used that is why I left it as int.

Commit:
Merge branch 'Implement-discount-and-discount-item-api's' into Implement-loyalty-api

gzili committed:
fix minor issues

gzili reviewed (commented):
Good job! Only some minor issues.
